api/v1/model_types.go

```diff
@@ -6,47 +6,42 @@ import (
 
 // ModelSpec defines the desired state of Model
 type ModelSpec struct {
-	// Source is a reference to the source of the model.
-	Source ModelSource `json:"source"`
-	// Training should be set to run a training job.
-	Training *ModelTraining `json:"training,omitempty"`
-	// Size describes different size dimensions of the underlying model.
-	Size ModelSize `json:"size"`
-	// Compute describes the compute requirements and preferences of the model.
-	Compute ModelCompute `json:"compute"`
-}
+	// Container that contains model code and dependencies.
+	Container Container `json:"container"`
+
+	// Loader should be set to run a loading job.
+	Loader *ModelLoader `json:"loader,omitempty"`
 
-type ModelCompute struct {
-	// +kubebuilder:validation:MinItems=1
-	// Types is a list of supported compute types for this Model. This list should be
-	// ordered by preference, with the most preferred type first.
-	Types []ComputeType `json:"types"`
+	// Trainer should be set to run a training job.
+	Trainer *ModelTrainer `json:"trainer,omitempty"`
 }
 
-// +kubebuilder:validation:Enum=CPU;GPU
-type ComputeType string
+func (m *Model) GetContainer() Container {
+	return m.Spec.Container
+}
+func (m *Model) SetContainer(c Container) {
```

Comment:
it doesn't look like SetContainer gets used anywhere?

Reply:
Removed